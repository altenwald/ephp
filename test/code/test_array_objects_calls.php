<?php

class A {
    function b($greet) { print "Says: [$greet] OK\n"; }
}

$hooks = [];
foreach (array ('a', 'b', 'c', 'd') as $filter) {
    if (!isset($hooks[$filter])) {
        $hooks[$filter] = new A;
    }
    $hooks[ $filter ]->b("world $filter");
}
