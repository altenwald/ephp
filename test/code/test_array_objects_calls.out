Says: [world a] OK
Says: [world b] OK
Says: [world c] OK
Says: [world d] OK
